tendrl/monitoring_integration/grafana/webhook_receiver.py

```diff
@@ -1,19 +1,72 @@
-from gevent.pywsgi import WSGIServer
-
+import gevent
+import gevent.greenlet
+import json
 
-def application(env, start_response):
-    if env['PATH_INFO'] == '/grafana_callback':
-        start_response('200 OK', [('Content-Type', 'text/html')])
-
-        # TODO(Rishubh) Receive Data and call parsing functions with alert json from grafana
-        # data = env['wsgi.input'].read()
+from gevent.pywsgi import WSGIServer
+from gevent.server import StreamServer
+from tendrl.commons.event import Event
+from tendrl.commons.message import ExceptionMessage
+from tendrl.monitoring_integration.alert.handlers import AlertHandlerManager
 
-        return [b"<b>Recieved</b>"]
 
-    start_response('404 Not Found', [('Content-Type', 'text/html')])
-    return [b'<h1>Not Found</h1>']
+class WebhookReceiver(gevent.greenlet.Greenlet):
+    def __init__(self):
+        super(WebhookReceiver, self).__init__()
+        self.server = WSGIServer(
+            ('127.0.0.1', 8789),
+            self.application
+        )
+        self.alert_handler = AlertHandlerManager()
+        
+    def application(self, env, start_response):
+        try:
+            if env['PATH_INFO'] != '/grafana_callback':
+                start_response(
+                    '404 Not Found',
```

Comment:
@anivargi @rishubhjain 404 response for alert with invalid PATH_INFO is ok or not?

Reply:
In my opinion the response is not of importance because we cannot actually receive and process on grafana side. Even if we don't send any response it's going to work the same way

Reply:
ya, I feel same here the response is not important, but without response, the WSGIServer will raise an exception so we need some response or need to send empty array atleast

Reply:
Sure